core-bundle/contao/library/Contao/DcaLoader.php

```diff
@@ -164,13 +164,11 @@ private function addDefaultLabels()
 					continue;
 				}
 
-				if (isset($GLOBALS['TL_LANG'][$this->strTable][$k]) || !isset($GLOBALS['TL_LANG']['DCA'][$k]))
-				{
-					$v['label'] = &$GLOBALS['TL_LANG'][$this->strTable][$k];
-				}
-				elseif (isset($GLOBALS['TL_LANG']['DCA'][$k]))
+				$v['label'] = &$GLOBALS['TL_LANG'][$this->strTable][$k];
+
+				if (!isset($GLOBALS['TL_LANG'][$this->strTable][$k]) && isset($GLOBALS['TL_LANG']['DCA'][$k]))
 				{
-					$v['label'] = &$GLOBALS['TL_LANG']['DCA'][$k];
+					$GLOBALS['TL_LANG'][$this->strTable][$k] = $GLOBALS['TL_LANG']['DCA'][$k];
```

Comment:
I don’t think it is a good idea to set values on `$GLOBALS['TL_LANG']` outside of our language files.

Reply:
what's _your language files_ in the DCA language loader? It is setting the labels to the _language files_ it loads the DCA for, and the if-condition skips if **exactly this** label is already set.

Reply:
In a DCA with a reference to `&$GLOBALS['TL_LANG']['tl_mytable']['something']` that is missing from the language file, with this change suddenly the label of `$GLOBALS['TL_LANG']['DCA']['something']` would show up. This would be very confusing and unexpected IMO.

I think changes to `$GLOBALS['TL_LANG']` should only be done in language files to prevent confusion and weird problems with references.

Reply:
It would also appear currently since we assign the label to the field? Also, this is exactly what we are trying to solve! If you did not assign an english label but only a german one, loading the DCA would replace the reference and now allow you to load german.

On another note, maybe that problem could be solved by using `array_key_exists` instead of `isset`?